Source/loadsave.cpp

```diff
@@ -1738,7 +1738,10 @@ static void SavePortal(SaveHelper *file, int i)
 void SaveHeroItems(PlayerStruct *pPlayer)
 {
 	size_t items = NUM_INVLOC + NUM_INV_GRID_ELEM + MAXBELTITEMS;
-	SaveHelper file("heroitems", items * sizeof(ItemStruct));
+	int itemSize = 368;

```

Comment:
The hardcode here is really unfortunate. Is there no way to get to this using proper `sizeof` calls?

Reply:
No, unfortunately not, the size is that of the original struct with x86 padding. But we can define it as a private const for the file so that it at least gets a name.

Reply:
Would it be possible to create a struct with fixed-size field types that replicated the original layout exactly, allowing us to use `sizeof` again?

I'd actually prefer that if it was possible and then mapping from one to the other.

Reply:
Technically I suppose, but it would rely on `pack 1` and a bunch of alignment files that go unused. But definatly not happening now for this.